posthog/hogql/placeholders.py

```diff
@@ -2,31 +2,35 @@
 
 from posthog.hogql import ast
 from posthog.hogql.errors import QueryError
+from posthog.hogql.utils import is_simple_value, deserialize_hx_ast
 from posthog.hogql.visitor import CloningVisitor, TraversingVisitor
 
 
-def replace_placeholders(node: ast.Expr, placeholders: Optional[dict[str, ast.Expr]]) -> ast.Expr:
-    return ReplacePlaceholders(placeholders).visit(node)
-
-
-def find_placeholders(node: ast.Expr) -> list[str]:
-    finder = FindPlaceholders()
-    finder.visit(node)
-    return list(finder.found)
-
-
 class FindPlaceholders(TraversingVisitor):
     def __init__(self):
         super().__init__()
-        self.found: set[str] = set()
+        self.has_expr_placeholders = False
+        self.has_filters = False
+        self.field_strings: set[str] = set()
 
     def visit_cte(self, node: ast.CTE):
         super().visit(node.expr)
 
     def visit_placeholder(self, node: ast.Placeholder):
-        if node.field is None:
-            raise QueryError("Placeholder expressions are not yet supported")
-        self.found.add(node.field)
+        field = node.field
+        if field:
+            if field == "filters" or field.startswith("filters."):
```

Comment:
a placeholder has a `chain` property now btw, so you can just do `chain[0] == 'filters'` instead